redfish_client/resource.py

```diff
@@ -109,6 +117,45 @@ def execute_action(self, action_name, payload):
             return self._connector.post(action.target, payload=payload)
         raise KeyError("Action with {} does not exist".format(action_name))
 
+    def wait_for_stat(self, stat, expected_value, fail_values=list(), poll_every_num_sec=3,
+                      timeout=30*60):
+        """
+        :param stat: list of keys or single key
+        :param expected_value: expected value
+        :param fail_values: fail values
+        :param poll_every_num_sec: number of requests per second
+        :param timeout: timeout in seconds
+        :return: actual value if expected_value matches else return None
+        Raises:
+            TimedOutError: If timeout is exceeded.
+        """
+
+        if "@odata.id" not in self._content:
+            raise Exception("Element does not have '@odata.id' attribute, cannot wait for a stat "
+                            "inside inner object")
+
+        start_time = time.time()
+        time_delta = 0
+
+        if not isinstance(stat, list):
+            stat = [stat]
+
+        while time_delta <= timeout:
+            self._clear_cache()
+
+            actual_value = self.dig(*stat)
+            if hasattr(actual_value, 'raw'):
+                actual_value = actual_value.raw
+
+            if actual_value == expected_value:
+                return actual_value
```

Comment:
Raising a custom exception sounds great to me too

Reply:
I changed it to return a tuple (boolean, value), so we also got an object in case we need it.

Reply:
Returning anything from this function is a bit pointless at this stage, since end-user cannot use this value in any meaningful way. If the wait call terminated successfully, we get back the same value that we were waiting for. If the wait failed, we do not return at all, since we raise an exception. What exactly went wrong can be determined by resource inspection in the `catch` part of the code, calling this function.